demos/shadow/shadow_demo_main/shadow_demo_main.c

```diff
@@ -0,0 +1,644 @@
+     * type is used for the dup, QoS, and retain flags. Hence masking
+     * out the lower bits to check if the packet is publish. */
+    if( ( pPacketInfo->type & 0xF0U ) == MQTT_PACKET_TYPE_PUBLISH )
+    {
+        assert( pDeserializedInfo->pPublishInfo != NULL );
+        LogInfo( ( "pPublishInfo->pTopicName:%s.\n\n", pDeserializedInfo->pPublishInfo->pTopicName ) );
+
+        /* Let the Device Shadow library tell us whether this is a device shadow message. */
+        if( SHADOW_SUCCESS == Shadow_MatchTopic( pDeserializedInfo->pPublishInfo->pTopicName,
+                                                 pDeserializedInfo->pPublishInfo->topicNameLength,
+                                                 & messageType,
+                                                 & pThingName,
+                                                 & thingNameLength ) )
+        {
+            /* Upon successful return, the messageType has been filled in. */
+            if( messageType == ShadowMessageTypeUpdateDelta )
+            {
+                /* Handler function to process payload. */
+                updateDeltaHandler( pDeserializedInfo->pPublishInfo );
+            }
+            else if ( messageType == ShadowMessageTypeUpdateAccepted )
+            {
+                /* Handler function to process payload. */
+                updateAcceptedHandler( pDeserializedInfo->pPublishInfo );
+            }
+            else if ( messageType == ShadowMessageTypeUpdateDocuments )
+            {
+                LogInfo( ( "/update/documents json payload:%s.\n\n", pDeserializedInfo->pPublishInfo->pPayload ) );
+            }
+            else if ( messageType == ShadowMessageTypeUpdateRejected )
+            {
+                LogInfo( ( "/update/rejected json payload:%s.\n\n", pDeserializedInfo->pPublishInfo->pPayload ) );
+            }
+            else
+            {
+                LogInfo( ( "Other message type:%d !!\n\n", messageType ) );
+            }
+        }
+        else
+        {
+            LogError( ( "Shadow_MatchTopic parse failed:%s !!\n\n", pDeserializedInfo->pPublishInfo->pTopicName ) );
+        }
+    }
+    else
+    {
+        handleOtherIncomingPacket( pPacketInfo, packetIdentifier );
+    }
+}
+
+/*-----------------------------------------------------------*/
+/**
+ * @brief Entry point of shadow demo.
+ *
+ * This main function demonstrates how to use the macros provided by the
+ * Device Shadow library to assemble strings for the MQTT topics defined
+ * by AWS IoT Device Shadow. It uses these macros for topics to subscribe
+ * to:
+ * - SHADOW_TOPIC_STRING_UPDATE_DELTA for "$aws/things/thingName/shadow/update/delta"
+ * - SHADOW_TOPIC_STRING_UPDATE_ACCEPTED for "$aws/things/thingName/shadow/update/accepted"
+ * - SHADOW_TOPIC_STRING_UPDATE_REJECTED for "$aws/things/thingName/shadow/update/rejected"
+ *
+ * It also used these macros for topics to publish to:
```

Comment:
used  --> uses

Reply:
will correct it.

Reply:
This [commit](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1104/commits/49b40ee1665d75b8ed186c326788b6200655e313) should fix the comment.
please let me know if you have other questions.